packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerViewDesktop.tsx

```diff
@@ -176,9 +176,19 @@ export function DateRangePickerViewDesktop<TDate>(inProps: DateRangePickerViewDe
   const isNextMonthDisabled = useNextMonthDisabled(currentMonth, { disableFuture, maxDate });
   const isPreviousMonthDisabled = usePreviousMonthDisabled(currentMonth, { disablePast, minDate });
 
+  // Range going for the start of the start day to the end of the end day.
+  // This makes sure that `isWithinRange` works with any time in the start and end day.
+  const valueDayRange = React.useMemo<DateRange<TDate>>(
```

Comment:
It seems to solve the root issue.
Should we still proceed with including this change in `alpha.2`?
If so, it should be replicated for Mobile version as well. 🤔

Reply:
Can't wait for #6416 to stop having this duplication...

Reply:
Done